src/evergreen/config.py

```diff
@@ -3,21 +3,82 @@
 from __future__ import absolute_import
 
 import os
-from collections import namedtuple
+import subprocess
 from typing import Dict, Optional
 
 import yaml
 
-EvgAuth = namedtuple("EvgAuth", ["username", "api_key"])
+
+class EvgAuth:  
+    def __init__(self, config_path: str, use_oauth: bool = True) -> None:
+        self.config_path: str = config_path
+        self.use_oauth: bool = use_oauth
+
+    def get_auth_headers(self) -> Dict[str, str | bytes]:
+        """Get the authentication headers for Evergreen API requests.
+
+        :return: Dictionary of headers with authentication information.
+        """
+        headers: Dict[str, str | bytes] = {}
+        if self.use_oauth:
+            headers["Authorization"] = f"Bearer {self.get_oauth_token()}"
+        else:
+            if not self.user:
+                conf = read_evergreen_from_file(self.config_path)
+                self.user: str = conf["user"]
+                self.api_key: str = conf["api_key"]
+            headers["Api-User"] = self.user
+            headers["Api-Key"] = self.api_key
+        return headers
+
+    def get_oauth_token(self) -> str:
+        """Get the OAuth token for authentication with Evergreen.
+
+        :return: OAuth token string.
+        """
+        process = subprocess.run(f"evergreen login --config ${self.config_path}", shell=True)
+        if process.returncode != 0:
+            raise RuntimeError("Failed to login to Evergreen using the provided config.")
+        process = subprocess.run(
+            f"evergreen client get-oauth-token --config {self.config_path}",
+            shell=True,
+            capture_output=True,
+        )
+        if process.returncode != 0:
+            raise RuntimeError("Failed to get OAuth token from Evergreen client.")
+        return process.stdout.decode("utf-8").strip()
+
 
 DEFAULT_NETWORK_TIMEOUT_SEC = 5 * 60
 DEFAULT_API_SERVER = "https://evergreen.mongodb.com"
+OAUTH_API_SERVER = "https://evergreen.corp.mongodb.com"
```

Comment:
Unfortunately Evergreen doesn't save this in the config in any capacity. This PR breaks [this](https://github.com/evergreen-ci/evergreen.py/pull/207/files) feature where it reads the URL from the config.

In Evergreen CLI, we do [this](https://github.com/ZackarySantana/evergreen/blob/58c84a0a573e878531c5272a92df565780df9d5f/operations/model.go#L271)- we don't respect the URL in the config for API requests.